hpi-studyu/studyu
# Fix/outdated cached subject

Cached `study_subject` could not be parsed after the `status` field for `Study` has been added. Now only the `progress` gets taken from the cache if there's any parsing issue and the remote subject will deliver all of the other data.

## How to test this PR

1. Start docker locally and make sure you dont have any containers running and all volumes are deleted.
1. Clone StudyU CLI, run `./studyu config` and only select Supabase. Then run `./studyu start` and follow the steps to launch a supabase instance. Copy the `ANON_KEY` shown at the end.
1. Open the studyu repository and run `flutter downgrade v3.19.6`. Then checkout the main branch, run `dart pub get` and `melos bs`. Replace the content of the file `flutter_common/lib/envs/.env` with this:

```
STUDYU_SUPABASE_URL=http://localhost:8082
STUDYU_SUPABASE_PUBLIC_ANON_KEY=<Your ANON_KEY>
STUDYU_PROJECT_GENERATOR_URL=
STUDYU_APP_URL=http://localhost:8080
STUDYU_DESIGNER_URL=http://localhost:8081
```

1. Start the Designer and create a study
1. Start the App with the android debugger
1. Complete some tasks
1. Check in [supabase studio](http://localhost:8082/) subject_progress table if data is inserted correctly.
1. Switch to [dev-update](https://github.com/hpi-studyu/studyu/tree/dev-update) branch
1. Run the contents of database/migration/20240526_migrate_close_study.sql in [supabase studio](http://localhost:8082/project/default/sql/1) sql execution page
1. Open the app again in the emulator (do not run the app again in your IDE! this is important as we want to test the state of the app of the main branch and not the dev state yet) and check if you can continue the study where you have left (also check if progress is stored inside supabase)
1. Upgrade to the newest version of flutter `flutter channel stable && flutter upgrade && dart pub get && melos bs`
1. Close the app in the emulator and build the new state of the dev branch by starting it through the IDE (This mimics the playstore update, so that we run the app in the dev state now)
1. Open the app and check if you can continue with the study
